### ISSUE TITLE
Content/Styles flicker during navigation on cloudflare pages & chrome

### ISSUE THREAD

**Issue by danthegoodman1** [withastro/starlight#2621]
### What version of `starlight` are you using?

0.29.2

### What version of `astro` are you using?

4.16.10

### What package manager are you using?

npm

### What operating system are you using?

mac

### What browser are you using?

chrome (I do not observe this on safari)

### Describe the Bug

See the flickering in this video (slowed down frame by frame)

https://github.com/user-attachments/assets/c0b17c52-337b-4646-b98a-c38ce9b8946f

Apologies for the low resolution, but you can see that on navigation it:
1. Drops the side nav
2. Takes 1 frame before it highlights the correct item in the right table of contents

I do not observe this behavior on https://starlight.astro.build/environmental-impact/ unless i haven't clicked on another page for a while, then I can observe it.

I don't observe this running locally, or on stackblitz, but I do see it on cloudflare pages. I've also tried incognito, clearing cache, etc. I also observe this just reloading the page, it seems page content renders in after

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-dwnlvm?file=tailwind.config.mjs <- I can't repro on this, but this is the same setup minus some md file content

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by danthegoodman1** [comment]
I also have updated chrome, still behaves this way

**Comment by danthegoodman1** [comment]
https://github.com/user-attachments/assets/32524d39-a816-4b51-befb-e66cce820b9a

I wonder if the behavior is linked to network speed? If i emulate a 3g network locally i can see a similar effect

**Comment by danthegoodman1** [comment]
I actually don't get this on a registered domain, just the .pages.dev domain...

**Comment by danthegoodman1** [comment]
bumping

**Comment by HiDeoo** [comment]
> I actually don't get this on a registered domain, just the .pages.dev domain...

That's quite interesting, and I'm personally not sure why that would be the case.

I uploaded the repro you shared on pages.dev and I cannot seem to reproduce the issue you're facing. Here is a video with no cache and network throttling to 3G:

https://github.com/user-attachments/assets/cefe5773-0f14-4d14-b435-dd9cb8225a10

Could you check if you're seeing the same issue on this page: https://starlight-issue-2621.pages.dev/guides/example/ ?

If not, would you be able to share a reproduction on pages.dev that showcases the issue you're facing?

**Comment by danthegoodman1** [comment]
@HiDeoo I get it on your provided example too

It's hard to record since it's so fast even at 3G, and sometimes the recording doesn't even capture it (I'm guessing bc my monitor is 144hz but recording is only 60fps)

**Comment by danthegoodman1** [comment]
Yeah it’s super weird, and I’ve got gigabit symmetric fiber with an m3 max chip so it’s def not a network or resource issue

**Comment by delucis** [comment]
Thanks for raising this @danthegoodman1 and for investigating @HiDeoo! We don’t seem to have a clear reproduction or obvious culprit for why this might be happening other than perhaps some slower connections or caching config specific to the hosting setup. Given that, I’m going to close this issue as something beyond our ability to influence at this time.
